### PR TITLE
Auto Logger

### PR DESCRIPTION
**Pull request by stieg**
Resolves issue #169 by adding speed based logging start and stop support that can be used tuned to their needs.  User must maintain X or above given speed for Y seconds to trigger a start event.  Conversely a user must maintain A or below speed for B seconds to trigger a stop event.

Still need to test but code compiles and all tests pass.

### CODE REVIEW

File: src/logger/auto_logger.c [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        json_float(serial, "speed", alst->speed, 2, true);
+        json_uint(serial, "time", alst->time, false);
+        json_objEnd(serial, more);
+}
+
+void auto_logger_get_config(struct auto_logger_config* cfg,
+                            struct Serial* serial,
+                            const bool more)
+{
+        json_objStartString(serial, "autoLoggerCfg");
```
Comment (brentpicasso) [inline src/logger/auto_logger.c]: Thoughts on how would this config structure be extensible for triggering on other sensors in the future?

File: src/gps/gpsTask.c [hunk 2]
```diff
@@ -64,6 +67,8 @@ void GPSTask(void *pvParameters)
                         if (result == GPS_MSG_SUCCESS) {
                                 const GpsSnapshot snap = getGpsSnapshot();
                                 lapstats_processUpdate(&snap);
+                                auto_logger_gps_sample_cb(&snap.sample);
```
Comment (brentpicasso) [inline src/gps/gpsTask.c]: I'd imagine in the future when triggering on other samples we'd move this somewhere central. loggerTask?

Comment (brentpicasso) [inline src/gps/gpsTask.c]: err... I meant, when triggering on other sensors we'd move it to a central location, like logger task.

File: src/logger/auto_logger.c (hunk 1 above)
Comment (stieg) [inline src/logger/auto_logger.c]: Discussed on Hangouts.  No change for now as anything more complex can be done in Lua

File: src/gps/gpsTask.c (hunk 2 above)
Comment (stieg) [inline src/gps/gpsTask.c]: Yeah... probably that or a timer task or something along those lines.

**Comment by stieg** [comment]
Hey @brentpicasso any further comments here?  I would like to merge is this soon since I have other patches that are built upon it that are waiting.

**Comment by brentpicasso** [comment]
@stieg I don't see any updates based on the conversation we had; where the auto logger state would maintain a separate logger state so that users can manually trigger a logging session and not have the autologger interfere.  Are you still going to add that?

**Comment by stieg** [comment]
@brentpicasso That would be because I forgot to push it.  Done now.

**Comment by brentpicasso** [comment]
Looks sane. I'm not in a place to do a real-world test. Did you @stieg  ?  If you did and works, I say :shipit:

**Comment by stieg** [comment]
Tested it last Friday using my RC Car test rig.  Worked wonderfully.

**Pull request merged**